sqlglot/optimizer/simplify.py

```diff
@@ -0,0 +1,100 @@
+import sqlglot.expressions as exp
+
+
+TRUE = exp.Boolean(this=True)
+FALSE = exp.Boolean(this=False)
+NULL = exp.Null()
+
+
+def simplify(expression):
+    expression = expression.copy()
+
+    # wrap because you cannot replace a node without a parent
+    expression = exp.Paren(this=expression)
+
+    while True:
+        start = hash(expression)
+        expression = simplify_equality(expression)
+        expression = simplify_not(expression)
+        expression = simplify_conjunctions(expression)
+        expression = simplify_parens(expression)
+
+        if start == hash(expression):
+            break
```

Comment:
I wonder if there could be an infinite loop scenario where one rule fights another. Cant think of one but just in case you can think of something

Reply:
yea, i thought of this, i couldn't think of a case where it happens yet because everything is a reduction